plugins/module_utils/resource.py

```diff
@@ -36,41 +47,68 @@ def resolve_refs(node, context):
     return node
 
 
-def run(desired_state, current_state, client, state):
-    sorter = TopologicalSorter()
-    for name, resource in desired_state.items():
-        if state == "present":
-            sorter.add(
-                name, *map(operator.itemgetter(1), REREG.findall(yaml.dump(resource)))
+class ResourceExceptionError(Exception):
+    def __init__(self, exc, msg):
+        self.exc = exc
+        self.msg = msg
+        super().__init__(self)
+
+
+class PravicCloudClient(metaclass=ABCMeta):
```

Comment:
This is great, my only comment is we should name this `CloudClient` so we don't have to rename it later once this collection gets a real name.